community/kernel/src/main/java/org/neo4j/kernel/impl/api/TwoPhaseNodeForRelationshipLocking.java

```diff
@@ -82,24 +81,48 @@ void lockAllNodesAndConsumeRelationships( long nodeId, final KernelStatement sta
         while ( retry );
     }
 
-    private void lockAllNodes( KernelStatement state )
+    private void collectAndSortNodeIds( long nodeId, KernelStatement state ) throws EntityNotFoundException
```

Comment:
Is it common to do detach delete for multiple nodes in one transaction? I'm thinking about the added primitive long set and array created for each such request and potential reuse of those data structures

Reply:
Hmm, I don't know how common that could be. If we start reusing objects here we should think about concurrency etc. so we store them in the correct place, which could be annoying. On the other hand, the created objects currently should be very short lived and probably thrown out on the first young GC. I think it will be fine to keep them.

Reply:
Yup, agree it may add complexity/other issues.